src/Compilers/Server/VBCSCompiler/ServerDispatcher.cs

```diff
@@ -231,13 +230,7 @@ private void ChangeToShuttingDown(string reason)
         private void RunGC()
         {
             _gcTask = null;
-            for (int i = 0; i < 10; i++)
-            {
-                GC.Collect();
-                GC.WaitForPendingFinalizers();
-            }
-            GCSettings.LargeObjectHeapCompactionMode = GCLargeObjectHeapCompactionMode.CompactOnce;
-            GC.Collect();
+            GC.GetTotalMemory(forceFullCollection: true);
```

Comment:
📝 It looks like my change doesn't preserve the forced compaction. It's not clear whether this is necessary. The easiest approach if this matterns might be to just set the compaction mode?

```suggestion
            GCSettings.LargeObjectHeapCompactionMode = GCLargeObjectHeapCompactionMode.CompactOnce;
            GC.GetTotalMemory(forceFullCollection: true);
```